### ISSUE TITLE
Removal Candidate

### ISSUE THREAD

**Issue by Adurol** [KaOS-Community-Packages/viber#1]
This Package will be removed in 3 days from KCP.

Reason: **[Blind copy](https://aur.archlinux.org/packages/viber/)**

Only submit packages that were adjusted for KaOS or you created yourself for KaOS. Blindly copying any existing PKGBUILD and submitting it to KCP is not the reason it was created for. If you can find working PKGBUILDs elsewhere, use them.

Please see also: [Using KaOS Community Packages](https://forum.kaosx.us/d/145-using-kaos-community-packages) and [KCP Guide](https://kaosx.us/docs/kcp/#submitting-a-package)

**Comment by padhia** [comment]
Have you compared the AUR and the KCP versions? What do you see is not adjusted for KaOS? I made sure it passed pckcp. What more is needed?

**Comment by Adurol** [comment]
Yes I compared the PKGBUILD here on KCP with the one on AUR. The files you submitted to KCP are with the same content, even the `eula.html` and `viber` are the same. And you stated even yourself in the commit message that it is from AUR. KCP is **not** about getting as much as possible PKGBULDs into it. The PKGBUILD in itself is alright, but there is no need to submit to KCP. It's as I wrote above: If the PKGBUILD you submit is practical identically to a PKGBUILD f.e. on AUR, CCR or somewhere else, don't submit it to KCP. The thing is more, that the PKGBUILD is not needed on KCP *because* it exists on AUR and works without (or only insignificant) modifications also on KaOS.

That a PKGBUILD passes pckcp is mandatory. Should it not pass, you will get a first time comment on it to correct it. For every following PKGBUILD that you submit that don't pass pckcp it will be silently deleted by the core team. I know it sound somewhat harsh, but to prevent KCP from overflowing with (broken) PKGBUILDs the core team made this ruling. KaOS is a small distro (and that is good that way) and KCP also only has a small active user base. If we don't abide this rules, we will only get broken packages and complaining users.

**Comment by padhia** [comment]
So what you are saying is, KaOS users should rely on AUR or CCR for software that they don't find in KCP, because KCP will contain PKGBUILDs only if AUR or CCR PKGBUILD are unable to be compiled correctly on KaOS. How does one figure that out? Is there a list of PKGBUILDs that KaOS users can safely obtain from AUR or CCR? Also, in such cases how does one use kcp's convenience of installing software from AUR and CCR?

Yes, I did say in github commit comment that it was from AUR. I also said that it was "Adapted" and not "Copied". I did take time to make sure the Arch dependencies are still valid in KaOS, create a README.md to direct users what dependencies are needed from KCP and cleaned up the PKGBUILD according to KCP guidelines. It just happened to be the case that PKGBUILD didn't need major changes -- apart from the KaOS conventions, but it still takes effort and time to figure that out too. I for one would have appreciated the fact that if a PKGBUILD exists in KCP, it's an indication that such software is being used some KaOS user and is more likely that it would build correctly than getting something from AUR or CCR. Since KaOS is a small distro, that's exactly the reason, good contributions should be welcomed to KCP.

And then there is your complaint about eula.html and viber being the same. eula.html is not something that I can "adjust" for KaOS (and I bet you, the AUR maintainer didn't come up with that either). If viber works the way it is, why change just to be different?

I understand that submissions to KCP needs to be policed for broken PKGBUILDs or the ones that are not likely to work on KaOS, such as 32 bit or Qt4 based. But removing a PKGBUILD just because it closely resembles AUR or CCR is being too aggressive.

**Comment by Adurol** [comment]
> So what you are saying is, KaOS users should rely on AUR or CCR for software that they don't find in KCP, because KCP will contain PKGBUILDs only if AUR or CCR PKGBUILD are unable to be compiled correctly on KaOS. How does one figure that out? Is there a list of PKGBUILDs that KaOS users can safely obtain from AUR or CCR?

No, there is no such list. You have to figure that out yourself. KaOS is for advanced users. The entry hurdle is low, because KaOS uses good tools such als Clamares and Octopi. But the distro itself is, in my opinion, not for inexperienced users. For me there are two ways to see it: Be happy with the software that you find in the repos and KCP (using GUI tools as inexperienced users) or be an advanced user and understand how pacman works, how to write PKGBUILDs and so on. There is hardly any middle ground in KaOS, sorry for that. That also is the reason for non-duplicate PKGBUILDs. As said, the goal of KCP is not to provide thousands of packages (the same goal as for the offical KaOS repos). If you want many packages, KaOS is not for you. Then something like Arch, Fedora or OpenSuse is better.

> Also, in such cases how does one use kcp's convenience of installing software from AUR and CCR?

KCP as "place to accumulate community packages" is pretty similar to AUR or CCR. But the tool used to access it (`kcp`) is different than f.e. yaourt. kcp will never do dependency checks or similar things that a package manger does. To say it a bit bluntly: kcp is just a download tool for PKGBUILDs on KCP and gives it then into `makepkg` to build them.

> Yes, I did say in github commit comment that it was from AUR. I also said that it was "Adapted" and not "Copied". I did take time to make sure the Arch dependencies are still valid in KaOS, create a README.md to direct users what dependencies are needed from KCP and cleaned up the PKGBUILD according to KCP guidelines. It just happened to be the case that PKGBUILD didn't need major changes -- apart from the KaOS conventions, but it still takes effort and time to figure that out too. I for one would have appreciated the fact that if a PKGBUILD exists in KCP, it's an indication that such software is being used some KaOS user and is more likely that it would build correctly than getting something from AUR or CCR. Since KaOS is a small distro, that's exactly the reason, good contributions should be welcomed to KCP.

> And then there is your complaint about eula.html and viber being the same. eula.html is not something that I can "adjust" for KaOS (and I bet you, the AUR maintainer didn't come up with that either). If viber works the way it is, why change just to be different?

If it sounded like a complaint, I apologize for that. I just wanted to point out the `eula.html` and `viber` to show what I wanted to say. (The PKGBUILD is not a 100% copy, as you said. But the only difference is the removed head and different 'pkgrel'.)

Please don't get me wrong. I welcome you contribution. Really. It's just that it doesn't align with the goal of KCP in this case. I understand you reasoning that users would appreciate this effort, but if we allow this reasoning for the majority of packages, than KCP will overflow with PKGBUILDs in no time. The result over time will be unmaintained or broken packages. If viber works out-of-the-box or after or f.e. renaming the dependencies because they are named different in KaOS than Arch, no need to submit it to KCP. "Everyone" can do this. If the user can't than sorry, KaOS is may be not the right choice in this case or there is the need to learn how the package manager in KaOS works to get to the point where one can do this.

Also, KaOS is really open source. It uses mostly well known things like calamares, octopi & pacman, github, etc. and the few custom tools are open source. Every user is free to create an alternative to KCP. There is for example the KCP-quarantine-area that ShalokShalom created. It's just that KCP is not the right place in this case and has different goals. What the community does is pretty much up the the community.

**Comment by padhia** [comment]
No need for an apology since I wasn't offended. Although it may sound counter-productive to me, but I guess you moderators have come up with guidelines that are in the best interests of the larger KaOS community. 

I am going to close this issue now. Feel free to remove this repo anytime -- no need to wait for the 3 day grace period. (I guess we'll never find out if viber was destined to be the most starred KCP repo :wink:)

Regards

**Comment by demmm** [comment]
I am not actively involved in KCP, but on this occasion I am with @padhi.

The text quoted was about blind copies (adding to KCP just what you get from AUR without looking at all what the PKGBUILD is).
In my opinion, KCP also needs to be about sharing PKGBUILDs so there is no need to hunt all the different places for needed files.
I think there still is no clear indication among those who moderate KCP what is acceptable and what is not.

If viber is not acceptable, then the whole KCP guide needs adjusting.  It encourages users to add to KCP,  just as long as they make the needed adjustments and check if it passes the test.

**Comment by Adurol** [comment]
Sorry for that, @padhia. It's as demm said, the whole concept of KCP is a bit foggy. We have some rules, but how to interpret them is still in the process. One thing is that once a package is in KCP, then it should be hard to remove it (even to the point that it wouldn't build for the time being). That's why I'm so strict in letting new packages in. Please be lenient with me on that. By no means I'm against submitting packages to KCP. If you have a PKGBUILD where you think it should be in KCP to let the community profit from it, please feel free to drop by in our IRC-channel or the forum. We will help you as much as possible.

**Comment by padhia** [comment]
No worries (and no need for sorry either). We are all learning and we all want results of our efforts to benefit the community as well. Sometime even with our best intentions that's not always the case, so do appreciate some watchful eyes looking out for the community, because I too, am part of that community.

I am sure under demm's guidance, KCP Core team will find a perfect equilibrium of letting useful contributions in while keeping the painful stuff out.

Good luck!
